android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt

```diff
@@ -54,11 +54,12 @@ suspend fun FhirEngine.loadLibraryAtPath(fhirOperator: FhirOperator, path: Strin
     runCatching { get<Library>(IdType(path).idPart) }.getOrNull()
       ?: search<Library> { filter(Library.URL, { value = path }) }.firstOrNull()
 
-  library?.let {
+  // TODO Fix refactored implementation on fhirOperator.loadLib()
+  /*library?.let {
     fhirOperator.loadLib(it)
 
     it.relatedArtifact.forEach { loadLibraryAtPath(fhirOperator, it) }
```

Comment:
can we remove these lines? if it's ok to comment them, why are they needed?

Reply:
I believe that can be addressed separately

Reply:
OK, but if this means we will not be able to evaluate measures, we can't merge this without the fix

Reply:
Yea. @allan-on can you tests this? I will also tag @maimoonak.

Reply:
Yea. @allan-on can you tests this? I will also tag @maimoonak.

Reply:
K thanks, and if we still can evaluate measures, then I think we can remove these lines until we need to use `loadLib` and address them then

Reply:
Removing these lines will break evaluation of measures. 

In summary, the changes on the SDK introduce loading of Resources, for instance, a Library that is used for evaluation of Immunizations, from the [KnowledgeManager](knowledge/src/main/java/com/google/android/fhir/knowledge/KnowledgeManager.kt). The KnowledgeManager persists them in a separate database and when we do evaluate any Measure there's a call to  the `knowledgeManager.loadResources()`. 

So in this specific implementation, the `loadLib()` was removed and instead there's [install(implementationGuide: ImplementationGuide, rootDirectory: File)](https://github.com/google/android-fhir/blob/master/knowledge/src/main/java/com/google/android/fhir/knowledge/KnowledgeManager.kt#L58 )
or [install(file: File)](https://github.com/google/android-fhir/blob/master/knowledge/src/main/java/com/google/android/fhir/knowledge/KnowledgeManager.kt#L76)

The refactors required on our end would probably need to make use of the provided implementations. 

From my understanding the resources required will already be available in the DB but in a different `ResourceEntity` table. 

cc @ellykits @pld @maimoonak

Reply:
OK, that's reasonable, and means we need to resolve this TODO before we can merge this PR

Reply:
The other alternative to get the rest of the changes in faster would be to hold off on updating the `workflow` dependency version, and then draft an implementation plan for using the `KnowledgeManager` for Knowledge Resources essentially packaged up in an Implementation Guide such as ValueSets, Plan/Activity Definitions and Library Resources containing CQL.

Reply:
Let's do that so we can get the other updates merged, then have a separate PR to update workflow and integrate that w/measure eval